### ISSUE TITLE
torch.norm documentation is misleading when it comes to matrix norms

### ISSUE THREAD

**Issue by nikitaved** [pytorch/pytorch#40033]
## 📚 Documentation

When it comes to parameter `dim` in `torch.norm`, the documentation says
```
dim (int, 2-tuple of python:ints, 2-list of python:ints, optional) – 
If it is an int, vector norm will be calculated, if it is 2-tuple of ints,
matrix norm will be calculated.
```

When `p` is specified and `dim` is a 2-tuple, I would expect `torch.norm` to compute an induced matrix norm, but instead it computes an L_p norm (vector norm) over a flattened matrix.

For example:
```python
In [1]: import torch

In [2]: x = torch.eye(3, 3)

In [3]: torch.norm(x, p=1, dim=(-2, -1)) # dim is a tuple, want a matrix norm
Out[3]: tensor(3.) # should be 1 in case of 1 induced matrix norm

```

cc @jlin27

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/12760

**Comment by ngimel** [comment]
Closing, this has been fixed.
